### ISSUE TITLE
Fail if no RPCO release version set

### ISSUE THREAD

**Issue by git-harry** [rcbops/rpc-openstack#1397]
###### Please provide the output to 'git describe --all' for  all deployments where you have confirmed it occurs.
heads/master

##### What type of issue is this? Please tick the appropriate box.
- [ ] bug - existing functionality that does not work for you as documented
- [x] feature - new functionality that you would like to see added

##### What is the expected behavior?
If the release version cannot be discovered, the RPCO playbooks should fail.

##### What is actually happening?
What currently happens is that the variable rpc_release gets set to 'UNKNOWN_RELEASE' if not the version is not discovered. Given that this should always be known it would be better to fail.

##### What were you attempting to do when you encountered this error (give as much detail as possible)? 
This was spotted as part of some MaaS work for Mitaka https://github.com/rcbops/u-suk-dev/issues/397

##### How can we reproduce this issue?
Once the fix for https://github.com/rcbops/u-suk-dev/issues/397 merges there should be nothing known that causes this issue on master. A test playbook that prints rpc_release should be sufficient to test this.

##### How can we verify your issue is addressed?
Attempting to use rpc_release when version has not been discovered fails.

##### If you are using or know of a workaround please describe the necessary steps?
There are no known breaking failures due to this, the workaround would depend on the failure.

##### What is your evaluation on the importance of this issue, and why?
While there are no know problems caused by this when the defined deployment/upgrade path is used it has the potential to cause unpredictable failures and so a prio-2 seems most reasonable.

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [ ] master
- [ ] mitaka-13.0
- [ ] liberty-12.2
- [ ] kilo

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [ ] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [ ] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by mattt416** [comment]
Speaking of this, would it be better to just hardcode the release into the code itself, like we do in [1]?  The upside being that we don't need different bits of code trying to auto-discover this, while the downside being that the release manager would have to keep this updated when new branches, etc. are created.  Thoughts?

[1] https://github.com/openstack/openstack-ansible/blob/stable/mitaka/playbooks/inventory/group_vars/all.yml#L17

**Comment by evrardjp** [comment]
I think it's better to have a script to bump version too.
This script could bump osa submodule sha, consuming roles sha (like ceph), change the appropriate version numbers, etc.

In the meantime, we can just ensure failures to avoid silent issues later.

**Comment by b3rnard0** [comment]
So not looking to blame anyone but I am looking for more context around why this work was completed. I see @cloader89 merged the issue based on some work by @major and @evrardjp, but as @git-harry mentioned, this has not been approved for work. Is there a specific customer drive or need that needs to be triaged or addressed? Thanks.

**Comment by cloader89** [comment]
No, I will take the blame here, I was just reviewing items in the PR list. I should have communicated with @evrardjp and I will revert it if necessary.

**Comment by b3rnard0** [comment]
@cloader89 Again, not looking to blame here just want a reason to make this work official or not. Just make sure that future work has been approved so that we can put our limited capacity to the things that are deemed important. 

@git-harry At this point, do we want to make this official and get this backported to the specific versions that need it? cc: @alextricity25

**Comment by b3rnard0** [comment]
@git-harry Let's pull this is at next week's replenishment meeting.
